### PR TITLE
Add tutorial Gotchas using NumPy

### PR DESCRIPTION
**Pull request by Ishitori**
## Description ##
Add tutorial Gotchas using NumPy. Explain about blocking calls and how to minimize the impact.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- [X] Code is well-documented: 
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Adds new tutorial

### CODE REVIEW

File: docs/tutorials/gluon/gotchas_numpy_in_mxnet.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+One key difference is in the way calculations are executed. Every `NDArray` manipulation in Apache MXNet is done in asynchronous, non-blocking way. That means, that when we write code like `c = a * b`, where both `a` and `b` are `NDArrays`, the function got pushed to the [Execution Engine](https://mxnet.incubator.apache.org/architecture/overview.html#execution-engine), which starts the calculation. The function immediately returns back, and the  user thread can continue execution, despite the fact that the calculation may not have been completed yet. 
+
+`Execution Engine` builds computation graph which may reorder or combine some calculations, but it honors dependency order: if there are other manipulation with `c` done later in the code, the `Execution Engine` will start doing them once the result of `c` is available. We don't need to write callbacks to start execution of subsequent code - the `Execution Engine` is going to do it for us.
```
Comment (larroy) [inline docs/tutorials/gluon/gotchas_numpy_in_mxnet.md]: builds *the*

File: docs/tutorials/gluon/gotchas_numpy_in_mxnet.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+Because of the blocking nature of [.asnumpy() method](https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#mxnet.ndarray.NDArray.asnumpy), using it reduces the execution performance, especially if the calculations are done on GPU: Apache MXNet has to copy data from GPU to CPU to return `np.array`. 
+
+The best solution is to **make manipulations directly on NDArrays by methods provided in [NDArray API](https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html)**.
+
+## NumPy operators vs. NDArray operators
+
+Despite the fact that [NDArray API](https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html) was specifically designed to be similar to `NumPy`, sometimes it is not easy to replace existing `NumPy` computations. The main reason is that not all operators, that are available in `NumPy`, are available in `NDArray API`. [This regularly updated page](https://github.com/apache/incubator-mxnet/issues/3199) contains the list of `NDArray API`  operators in progress, where:
```
Comment (rahul003) [inline docs/tutorials/gluon/gotchas_numpy_in_mxnet.md]: The page linked doesn't seem to be updated in 2 years. Is that the only reference we can provide? We should probably at least remove the words 'regularly updated'

**Comment by sandeep-krishnamurthy** [comment]
@aaronmarkham - ping

**Comment by sandeep-krishnamurthy** [comment]
@lshitori - Can you please address few suggested changes by @aaronmarkham and this is all ready to go!

**Comment by Ishitori** [comment]
@sandeep-krishnamurthy @aaronmarkham @larroy @rahul003  thanks for the review. Updated the tutorial based on your comments

**Pull request merged**